lib/utils.js

```diff
@@ -9,128 +9,136 @@ const parseJson = require('parse-json');
-    assert(
-        Array.isArray(feedIds) && feedIds.length > 0,
-        `Expected at least 1 feed id, but got ${feedIds}`
-    );
+    async function bundleAndUpload({ sink, type, feedIds, uri, ...options }) {
+        assert(
+            Array.isArray(feedIds) && feedIds.length > 0,
+            `Expected at least 1 feed id, but got ${feedIds}`
+        );
 
-    const fetchedFeeds = await fetchFeeds(sink, feedIds, 3);
-    const parsedFeeds = parseFeedContent(fetchedFeeds);
-    const content = await bundleFeeds(parsedFeeds, type, options);
-    const fileName = `${hasher(content)}.${type}`;
-    await upload(sink, fileName, content, 3);
+        const fetchedFeeds = await fetchFeeds(sink, feedIds, 3);
+        const parsedFeeds = parseFeedContent(fetchedFeeds);
+        const content = await bundleFeeds(parsedFeeds, type, options);
+        const fileName = `${hasher(content)}.${type}`;
+        await upload(sink, fileName, content, 3);
 
-    return {
-        file: fileName,
-        uri: uri + fileName,
-    };
-}
+        return {
+            file: fileName,
+            uri: uri + fileName,
+        };
+    }
 
-async function parseBody(req, res) {
-    try {
-        return await body(req, res, {});
-    } catch (e) {
-        throw Boom.boomify(e, {
-            statusCode: 400,
-            message:
-                'Unparsable feed data given in POST-body. Invalid or empty JSON payload.',
-        });
+    async function parseBody(req, res) {
+        try {
+            return await body(req, res, {});
+        } catch (e) {
+            throw Boom.boomify(e, {
+                statusCode: 400,
+                message:
+                    'Unparsable feed data given in POST-body. Invalid or empty JSON payload.',
+            });
+        }
     }
-}
 
-function validateFeeds(feeds) {
-    const result = schemas.ids.validate(feeds);
-    if (result.error) {
-        throw Boom.boomify(result.error, {
-            statusCode: 400,
-            message: 'Invalid feed data given in POST-body.',
-        });
+    function validateFeeds(feeds) {
+        const result = schemas.ids.validate(feeds);
+        if (result.error) {
+            throw Boom.boomify(result.error, {
+                statusCode: 400,
+                message: 'Invalid feed data given in POST-body.',
+            });
+        }
+        return result.value;
     }
-    return result.value;
-}
 
-function booleanWithDefault(value, defaultValue) {
-    if (value === 'true' || value === true) return true;
-    if (value === 'false' || value === false) return false;
-    return defaultValue;
-}
+    function booleanWithDefault(value, defaultValue) {
+        if (value === 'true' || value === true) return true;
+        if (value === 'false' || value === false) return false;
+        return defaultValue;
+    }
 
-function endWorkers() {
-    bundleWorker.end();
-}
+    function endWorkers() {
+        bundler.end();
```

Comment:
`end` is only available when it's a worker, need to check here

Reply:
true